### PR TITLE
[OPIK-374] Add bulk actions to all tables

### LINKED ISSUE #494: [FR]: Add multi-select options for Dataset view
### Willingness to contribute

No. I can't contribute this feature at this time.

### Proposal summary

It would be great if the experiment/dataset/projets pages UI is consistent. At the moment you can only delete one dataset at the time.

### Motivation

I have created 13 datasets which i don't want to keep. It's annoying to manually delete each

### PR DESCRIPTION
**Pull request by andriidudar**
## Details
image
image
image
image

## Issues

Resolves #494

### CODE REVIEW

File: apps/opik-frontend/src/components/pages/ProjectsPage/ProjectsActionsPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const disabled = !projects?.length;
+
+  const { mutate } = useProjectBatchDeleteMutation();
+
+  const deleteProjectsHandler = useCallback(() => {
+    mutate({
+      ids: projects.map((p) => p.id),
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [projects]);
```
Comment (aadereiko) [inline apps/opik-frontend/src/components/pages/ProjectsPage/ProjectsActionsPanel.tsx]: you could just add `mutate` here. it's not as heavy as the entire `useProjectBatchDeleteMutation()`

File: apps/opik-frontend/src/components/pages/FeedbackDefinitionsPage/FeedbackDefinitionsActionsPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const disabled = !feedbackDefinitions?.length;
+
+  const { mutate } = useFeedbackDefinitionBatchDeleteMutation();
+
+  const deleteFeedbackDefinitionsHandler = useCallback(() => {
+    mutate({
+      ids: feedbackDefinitions.map((f) => f.id),
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [feedbackDefinitions]);
```
Comment (aadereiko) [inline apps/opik-frontend/src/components/pages/FeedbackDefinitionsPage/FeedbackDefinitionsActionsPanel.tsx]: i would add `mutate` to the dependencies

File: apps/opik-frontend/src/components/pages/DatasetsPage/DatasetsActionsPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const disabled = !datasets?.length;
+
+  const { mutate } = useDatasetBatchDeleteMutation();
+
+  const deleteDatasetsHandler = useCallback(() => {
+    mutate({
+      ids: datasets.map((d) => d.id),
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [datasets]);
```
Comment (aadereiko) [inline apps/opik-frontend/src/components/pages/DatasetsPage/DatasetsActionsPanel.tsx]: i would add `mutate` to the list of dependencies

File: apps/opik-frontend/src/components/pages/DatasetItemsPage/DatasetItemsActionsPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  const disabled = !datasetItems?.length;
+
+  const { mutate } = useDatasetItemBatchDeleteMutation();
+
+  const deleteDatasetItemsHandler = useCallback(() => {
+    mutate({
+      ids: datasetItems.map((i) => i.id),
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [datasetItems]);
```
Comment (aadereiko) [inline apps/opik-frontend/src/components/pages/DatasetItemsPage/DatasetItemsActionsPanel.tsx]: i would add `mutate` to the dependencies

**Review by aadereiko — CHANGES_REQUESTED** [review]
Is it supposed to have such a big margin?

image


image


in Experiments it looks nice:
image

File: apps/opik-frontend/src/components/pages/ProjectsPage/ProjectsActionsPanel.tsx (hunk 1 above)
Comment (andriidudar) [reply apps/opik-frontend/src/components/pages/ProjectsPage/ProjectsActionsPanel.tsx]: Fixed

File: apps/opik-frontend/src/components/pages/FeedbackDefinitionsPage/FeedbackDefinitionsActionsPanel.tsx (hunk 2 above)
Comment (andriidudar) [reply apps/opik-frontend/src/components/pages/FeedbackDefinitionsPage/FeedbackDefinitionsActionsPanel.tsx]: Fixed

File: apps/opik-frontend/src/components/pages/DatasetsPage/DatasetsActionsPanel.tsx (hunk 3 above)
Comment (andriidudar) [reply apps/opik-frontend/src/components/pages/DatasetsPage/DatasetsActionsPanel.tsx]: Fixed

File: apps/opik-frontend/src/components/pages/DatasetItemsPage/DatasetItemsActionsPanel.tsx (hunk 4 above)
Comment (andriidudar) [reply apps/opik-frontend/src/components/pages/DatasetItemsPage/DatasetItemsActionsPanel.tsx]: Fixed

**Comment by andriidudar** [comment]
> Is it supposed to have such a big margin?
> 
> image image
> in Experiments it looks nice: image

It is a known issue, it will be fixed in one of the next PRs.

**Review by aadereiko — APPROVED** [review]

**Pull request merged**